contracts/bonding/libraries/TranscoderPool.sol

```diff
@@ -0,0 +1,372 @@
+                // Set prev pointer of next node to the previous node
+                self.nodes[self.nodes[_transcoder].nextTranscoder].prevTranscoder = self.nodes[_transcoder].prevTranscoder;
+            }
+        } else {
+            // List contains a single node
+            // Set the head and tail to null
+            self.head = address(0);
+            self.tail = address(0);
+        }
+
+        delete self.nodes[_transcoder];
+        self.size--;
+    }
+
+    /*
+     * @dev Increase the stake of a transcoder in the list
+     * @param _transcoder Address of transcoder
+     * @param _amount Amount to increase transcoder's stake by
+     * @param _worseTranscoder Address of a transcoder with less stake than the updated transcoder
+     * @param _betterTranscoder Address of a transcoder with more stake than the updated transcoder
+     */
+    function increaseTranscoderStake(Data storage self, address _transcoder, uint256 _amount, address _worseTranscoder, address _betterTranscoder) public {
+        // List must contain the transcoder
+        require(contains(self, _transcoder));
+
+        // Compute new stake
+        uint256 newStake = self.nodes[_transcoder].stake.add(_amount);
+        // Remove transcoder from the list
+        removeTranscoder(self, _transcoder);
+        // Insert transcoder into the list with its new stake
+        addTranscoder(self, _transcoder, newStake, _worseTranscoder, _betterTranscoder);
+    }
+
+    /*
+     * @dev Decrease the stake of a transcoder in the list
+     * @param _transcoder Address of transcoder
+     * @param _amount Amount to decrease transcoder's stake by
+     * @param _worseTranscoder Address of a transcoder with less stake than the updated transcoder
+     * @param _betterTranscoder Address of a transcoder with more stake than the updated transcoder
+     */
+    function decreaseTranscoderStake(Data storage self, address _transcoder, uint256 _amount, address _worseTranscoder, address _betterTranscoder) public {
+        // List must contain the transcoder
+        require(contains(self, _transcoder));
+
+        // Compute new stake
+        uint256 newStake = self.nodes[_transcoder].stake.sub(_amount);
+        // Remove transcoder from the list
+        removeTranscoder(self, _transcoder);
+
+        if (newStake > 0) {
+            // Only add transcoder if its new stake is non-zero
+            // Insert transcoder into the list with its new stake
+            addTranscoder(self, _transcoder, newStake, _worseTranscoder, _betterTranscoder);
+        }
+    }
+
+    /*
+     * @dev Checks if the list contains a transcoder
+     * @param _transcoder Address of transcoder
+     */
+    function contains(Data storage self, address _transcoder) public view returns (bool) {
+        return self.head == _transcoder || self.tail == _transcoder || self.nodes[_transcoder].prevTranscoder != address(0) || self.nodes[_transcoder].nextTranscoder != address(0);
```

Comment:
Can we just check the mapping to see if the returned node for this transcoder address has non-zero stake?